src/util/fast-serialize.ts

```diff
@@ -0,0 +1,204 @@
+/**
+ * Fast serialization module for EA response objects
+ *
+ * This module provides optimized JSON serialization for common EA response patterns,
+ * significantly reducing CPU usage, GC pressure, and improving response times
+ * under high load scenarios.
+ */
+
+// Import required types and metrics
+import { AdapterResponse, ResponseGenerics } from './types'
+import { metrics } from '../metrics'
+
+/**
+ * Fast path response serialization
+ *
+ * This implementation uses specialized object detection and hand-optimized
+ * JSON string construction for common response patterns, avoiding the overhead
+ * of the full JSON.stringify process. For complex or unusual responses, it
+ * falls back to standard JSON.stringify.
+ *
+ * @param response - The response object to serialize
+ * @returns JSON string representation of the response
+ */
+export function serializeResponse<T extends ResponseGenerics>(
```

Comment:
I'd be really curious to A/B test this feature, as the JSON.stringify call even tho slow it's using a c lib iirc, but probably for small payloads the string concatenation is faster? It seems somewhat brittle to me though, it's good to try but the approach should be stricter if it turns out it gives significant benefits